test/ipns.js

```diff
@@ -0,0 +1,130 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const series = require('async/series')
+const parallel = require('async/parallel')
+const os = require('os')
+const path = require('path')
+const hat = require('hat')
+
+const DaemonFactory = require('ipfsd-ctl')
+
+const spawnJsDaemon = (dir, callback) => {
+  DaemonFactory.create({ type: 'js' })
+    .spawn({
+      repoPath: dir,
+      disposable: false,
+      initOptions: { bits: 512 }
+    }, callback)
+}
+
+const spawnGoDaemon = (dir, callback) => {
+  DaemonFactory.create()
+    .spawn({
+      repoPath: dir,
+      disposable: false,
+      initOptions: { bits: 1024 },
+      args: ['--offline']
+    }, callback)
+}
+
+const ipfsRef = '/ipfs/QmPFVLPmp9zv5Z5KUqLhe2EivAGccQW2r7M7jhVJGLZoZU'
+
+const publishAndResolve = (publisherDaemon, resolverDaemon, callback) => {
+  let nodeId
+  let sameDaemon = false
+
+  if (typeof resolverDaemon === 'function') {
+    callback = resolverDaemon
+    resolverDaemon = publisherDaemon
+    sameDaemon = true
+  }
+
+  const stopPublisherAndStartResolverDaemon = (callback) => {
+    if (sameDaemon) {
+      return callback()
+    }
+    series([
+      (cb) => publisherDaemon.stop(cb),
+      (cb) => setTimeout(cb, 2000),
+      (cb) => resolverDaemon.start(cb)
+    ], callback)
+  }
+
+  series([
+    (cb) => publisherDaemon.init(cb),
+    (cb) => publisherDaemon.start(cb),
+    (cb) => publisherDaemon.api.id((err, res) => {
+      expect(err).to.not.exist()
+      nodeId = res.id
+      cb()
+    }),
+    (cb) => publisherDaemon.api.name.publish(ipfsRef, { resolve: false }, cb),
+    (cb) => stopPublisherAndStartResolverDaemon(cb),
+    (cb) => {
+      resolverDaemon.api.name.resolve(nodeId, { local: true }, (err, res) => {
+        expect(err).to.not.exist()
+        expect(res).to.equal(ipfsRef)
+        cb()
+      })
+    },
+    (cb) => resolverDaemon.stop(cb),
+    (cb) => setTimeout(cb, 2000),
+    (cb) => resolverDaemon.cleanup(cb)
```

Comment:
Are we missing cleanup for the other daemon?

Reply:
The cleanup is for the repo. 

I think that you did not understand the flow here, considering this comment and the next one. The flow is the following:

1) Init X daemon (with repo Z)
2) Start X daemon
3) IPNS Publish
4) Stop X daemon
5) Start Y daemon (with repo Z)
6) IPNS Resolve
7) Stop Y daemon
8) Clean repo Z

X and Y may be replaced for JS or GO.

This way, both implementations will use the same repo to publish and resolve the IPNS record, since we do not have routing yet.

Reply:
Ah, yes I realised what was going on later and forgot to remove these comments. Please ignore!